extensions/amp-story/1.0/media-pool.js

```diff
@@ -585,28 +500,32 @@ export class MediaPool {
       )
     );
     poolMediaEl[REPLACED_MEDIA_PROPERTY_NAME] = null;
-
-    const swapOutOfDom = this.enqueueMediaElementTask_(
-      poolMediaEl,
-      new SwapOutOfDomTask(placeholderEl)
-    );
-
+    swapMediaOutOfDom(poolMediaEl, placeholderEl);
     this.resetPoolMediaElementSource_(poolMediaEl);
-    return swapOutOfDom;
+  }
+
+  /**
+   * @param {!PoolBoundElementDef} poolMediaEl
+   * @private
+   */
+  resetPoolMediaElementSource_(poolMediaEl) {
+    // TODO(wg-stories): This should be a simpler Sources.removeFrom() call.
+    // Test current use, and update.
+    updateSources(this.win, poolMediaEl, new Sources());
```

Comment:
Plz bring back the `load()` call on the `poolMediaEl`. If you set the sources but don't call `load()`, the readyState or buffered frames aren't reset, and it will trip our video load promises when we re-use the element.

(Side note that this is why I'm so scared of this refactoring, it's super easy to introduce regressions that we won't notice)